block/block.go

```diff
@@ -41,6 +47,11 @@ func (m *Manager) applyBlock(ctx context.Context, block *types.Block, commit *ty
 		return err
 	}
 
+	if m.lastState.IsGenesis() {
```

Comment:
also for this. will be probably done once so maybe there is a a better place.
also not sure about the scenario we're checking - comments would be helpful also here.